apps/dcos_net/src/dcos_net_mesos.erl

```diff
@@ -18,23 +19,47 @@ poll(URIPath) ->
     handle_response(Response).
 
 -spec(call(jiffy:json_term()) ->
-    {ok, jiffy:json_term()} | {error, Reason :: term()}).
+    {ok, jiffy:json_term()} | {ok, reference(), pid()} | {error, term()}).
 call(Request) ->
     call(Request, [], []).
 
 -spec(call(jiffy:json_term(), httpc:http_options(), httpc:options()) ->
     {ok, jiffy:json_term()} | {ok, reference(), pid()} | {error, term()}).
 call(Request, HTTPOptions, Opts) ->
+    Type = maps:get(type, Request),
+    Begin = erlang:monotonic_time(),
     ContentType = "application/json",
     HTTPRequest = {"/api/v1", [], ContentType, jiffy:encode(Request)},
     Opts0 = [{sync, false}|Opts],
     {ok, Ref} = request(post, HTTPRequest, HTTPOptions, Opts0),
+    try wait_for_response(Ref) of
+        {ok, Ref, Pid} ->
+            {ok, Ref, Pid};
+        {ok, Data} ->
+            prometheus_counter:inc(
```

Comment:
does it make sense to have `call_success_total` as well?

Reply:
There is a metric for how many requests dcos-net did. It's a part of `call_duration_seconds` summary (in prometheus, it will be two metrics, `call_duration_seconds_count` and `call_duration_seconds_sum`). `call_success_total` is equal to `call_duration_seconds_count - call_failures_total`.